### ISSUE TITLE
[enhancement] Allow app add to specify the names of the deployments

### ISSUE THREAD

**Issue by pke** [Microsoft/code-push#272]
Currently it does not seem possible to specify the default deployments when adding an app like this:
`code-push app add appname-android development production staging`

Would be nice if I would not have to rename the default deployments after I add an app.

**Comment by lostintangent** [comment]
Yeah I agree with you about this. Aside from being able to add new default deployments ("development" in your case), what if the deployment name resolution was changed to be case-insensitive? That way, the existing staging and production deployments can be targeted using lowercase if you preferred (e.g. "code-push deployment h foo staging")? Would that be valuable? Or do you want the deployment names to be explicitly lowercase?

**Comment by pke** [comment]
case insensitive handling would also be a nice addition to this feature request.

**Comment by pke** [comment]
Is this being worked on?

**Comment by pke** [comment]
Is there any way this could be contributed by the community? Guess it isn't since the server is still closed source?

**Comment by silhouettes** [comment]
A PR can definitely be made! The `app add` command is [here](https://github.com/Microsoft/code-push/blob/master/cli/script/command-executor.ts#L164). One way is to programmatically [delete the deployments](https://github.com/Microsoft/code-push/blob/master/cli/script/command-executor.ts#L349) after they're added, but the server already supports turning off automatic deployment creation, you'd just need to modify the management SDK slightly [here](https://github.com/Microsoft/code-push/blob/master/sdk/script/management-sdk.ts#L210):

```
    public addApp(appName: string, manuallyProvisionDeployments: boolean = false): Promise<App> {
        var app: App = { name: appName, manuallyProvisionDeployments: manuallyProvisionDeployments };
```

And after that you can add custom deployments as needed [like this](https://github.com/Microsoft/code-push/blob/master/cli/script/command-executor.ts#L289).

Our SDK is extremely flexible and can be used to do almost anything you can do with the CLI.

**Comment by panarasi** [comment]
Once Code Push is merged into Mobile Center, it is easier to create deployments in the UI. Hence, closing this issue. If you still think that it makes sense to add this functionality in CodePush, please do let us know.
